cdap-hbase-compat-0.94/src/main/java/co/cask/cdap/data2/util/hbase/HBase94TableUtil.java

```diff
@@ -21,24 +21,78 @@
 import co.cask.cdap.data2.transaction.coprocessor.hbase94.DefaultTransactionProcessor;
 import co.cask.cdap.data2.transaction.queue.coprocessor.hbase94.DequeueScanObserver;
 import co.cask.cdap.data2.transaction.queue.coprocessor.hbase94.HBaseQueueRegionObserver;
+import co.cask.cdap.proto.Id;
+import com.google.common.base.Joiner;
 import com.google.common.collect.Maps;
+import org.apache.hadoop.conf.Configuration;
 import org.apache.hadoop.hbase.ClusterStatus;
 import org.apache.hadoop.hbase.Coprocessor;
 import org.apache.hadoop.hbase.HColumnDescriptor;
 import org.apache.hadoop.hbase.HRegionInfo;
 import org.apache.hadoop.hbase.HServerLoad;
+import org.apache.hadoop.hbase.HTableDescriptor;
 import org.apache.hadoop.hbase.ServerName;
 import org.apache.hadoop.hbase.client.HBaseAdmin;
+import org.apache.hadoop.hbase.client.HTable;
 import org.apache.hadoop.hbase.io.hfile.Compression;
 import org.apache.hadoop.hbase.regionserver.StoreFile;
 
 import java.io.IOException;
 import java.util.Map;
+import javax.annotation.Nullable;
 
 /**
  *
  */
 public class HBase94TableUtil extends HBaseTableUtil {
+
+  @Override
+  public boolean namespacesSupported() {
+    return false;
+  }
+
+  @Override
+  public HTable getHTable(Configuration conf, @Nullable String namespace, String tableName) throws IOException {
+    // TODO: Think about what happens if namespace passed here is null.maybe it should never be null.
+    return new HTable(conf, getActualTableName(prefixCDAPToNamespace(namespace), tableName));
+  }
+
+  @Override
+  public HTableDescriptor getHTableDescriptor(@Nullable String namespace, String tableName) {
+    // TODO: Think about what happens if namespace passed here is null.maybe it should never be null.
+    return new HTableDescriptor(getActualTableName(prefixCDAPToNamespace(namespace), tableName));
+  }
+
+  @Override
+  public boolean hasNamespace(HBaseAdmin admin, Id.Namespace namespace) {
+    return true;
+  }
+
+  @Override
+  public void createNamespace(HBaseAdmin admin, Id.Namespace namespace) throws IOException {
+    // No-op.
+  }
+
+  @Override
+  public void deleteNamespace(HBaseAdmin admin, Id.Namespace namespace) throws IOException {
+    // No-op
+  }
+
+  @Override
+  public String getTableNameWithNamespace(String namespace, String tableName) {
```

Comment:
Actually on second thoughts, I think I'll keep it for now, because I see usages in ``HBaseStreamFileConsumerFactory`` for example that create a new ``HBaseAdmin`` and use methods like ``disableTable(String tablename)``. We may eventually want to move these methods to the compat module. I'll see if I can do that during namespacing but we'll have to keep these till then.